packages/form-builder/src/blocks/fields/field.js

```diff
@@ -1,6 +1,73 @@
 import {Icon} from '@wordpress/icons';
 import {__} from "@wordpress/i18n";
 import settings from "./settings";
+import {useFieldNames} from "../../hooks";
+import {InspectorAdvancedControls} from "@wordpress/block-editor";
+import {ExternalLink, PanelRow, TextControl} from "@wordpress/components";
+
+const Edit = (props) => {
+    const ParentEdit = settings.edit
+
+    const {attributes: {fieldName, label}, setAttributes} = props;
+
+    const validateFieldName = useFieldNames()
+
+    const slugify = (value) => {
+        return value.toLowerCase()
+                    .replace(/\s|_/g, '-') // Replace spaces and underscores with dashes
+                    .replace(/[^a-zA-Z\d\s-]/g, '') // Replace non-alphanumeric characters (other than dashes)
+                    .replace(/-$/g, '') // Remove trailing dash
+    }
```

Comment:
This can be extracted outside the component, let's throw it in the common folder and pull it in here.

Reply:
resolved f5a47a87